### PR TITLE
Add Go 1.4 to travis and sync the test-go with origin

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@soltysh PTAL

(I added golint there as an optional test, just want to see if the script itself is working might be useful in future)

**Comment by soltysh** [comment]
LGTM, but I'd rather not to have golint yet, since I know we're not ready for it :wink:

**Comment by mfojtik** [comment]
@soltysh my intent was to just exercise it so we know it 'works' ;-) (have to be fixed for travis, but we can do that later).

**Comment by mfojtik** [comment]
[merge]

**Comment by mfojtik** [comment]
[merge]

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,17 @@ language: go
...
-  - go get code.google.com/p/go.tools/cmd/cover
+  - go get golang.org/x/tools/cmd/cover
+  - go get -u github.com/golang/lint/golint
+  - ./hack/verify-gofmt.sh
+  - ./hack/verify-golint.sh || true
   - ./hack/build-go.sh
 
 script:
-  - ./hack/test-go.sh
+  - KUBE_RACE="-race" KUBE_COVER="-cover -covermode=atomic" ./hack/test-go.sh "" -p=4
```
Comment (bparees) [inline .travis.yml]: seems odd our scripts use KUBE env variables....

Comment (soltysh) [inline .travis.yml]: Actually we're not, I've renamed all of them to be `STI_*`, PR with fix to above https://github.com/openshift/source-to-image/pull/141

**Pull request merged**